app/models/solidus_stripe/webhook_endpoint.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module SolidusStripe
+  # Represents a webhook endpoint for a {SolidusStripe::PaymentMethod}.
+  #
+  # A Stripe webhook endpoint is a URL that Stripe will send events to. A store
+  # could have multiple Stripe payment methods (e.g., a marketplace), so we need
+  # to differentiate which one a webhook request targets.
+  #
+  # This model associates a suffix with a payment method. The suffix is appended
+  # to the endpoint URL (`.../webhooks/:suffix`) so that we can fetch the
+  # correct payment method from the database and bind it to the generated
+  # `Spree::Bus` event.
+  #
+  # We use a suffix instead of the payment method ID to be resilient to
+  # database changes and to avoid guessing about valid endpoint URLs.
+  class WebhookEndpoint < ::Spree::Base
+    belongs_to :payment_method,
+      class_name: 'SolidusStripe::PaymentMethod'
+
+    # @api private
+    def self.generate_suffix
+      SecureRandom.hex(16).then do |suffix|
+        exists?(suffix: suffix) ? generate_suffix : suffix
```

Comment:
What about also enforcing this with a validation? Also, we could consider adding a unique index on the database table, both for performances and to enforce this validation on the db level. Thoughts?

Reply:
> What about also enforcing this with a validation? 

I think there's no need to add validation for that, as that's not part of user input. The library code generates that, so testing is the right place to "validate" it. What do you think?

> Also, we could consider adding a unique index on the database table, both for performances and to enforce this validation on the db level. Thoughts?

I agree. I added the index and made it unique. Probably we won't appreciate any performance difference (very few records for `Spree::PaymentMethod), and, in fact, making the index unique will impact negatively, but I like to have consistency at the DB layer unless there's a reason to skip. Thanks for your suggestion.